docs/v0.3/sdk/go/8g34z-get-started.md

```diff
@@ -0,0 +1,452 @@
+# Get Started with the Dagger SDK for Go
+
+## Introduction
+
+[Dagger](https://dagger.io/) is a DevOps automation engine. It enables DevOps teams to orchestrate and deliver pipelines using any supported programming language. Popular use cases include multi-stage builds, CI/CD, on-demand staging environments, machine learning pipelines, end-to-end testing, artifact scanning and signing, and more.
+
+Dagger 0.3 exposes its functionality via a low-level language-agnostic API framework and provides native language SDKs to communicate with this API. The Dagger SDK for Go is one such SDK. It enables pipeline authors to use Dagger directly from Go, without needing to learn a new language.
+
+This tutorial teaches you the basics of using Dagger in Go. You will learn how to:
+
+- Install the Dagger SDK for Go
+- Run and test a simple application with Dagger
+- Build and publish a container image of the application with Dagger
+
+## Requirements
+
+This tutorial assumes that:
+
+- You have a basic understanding of the Go programming language. If not, [read the Go tutorial](https://go.dev/doc/tutorial/getting-started).
+- You have a Go development environment with Go 1.15 or later. If not, [download and install Go](https://go.dev/doc/install).
+- You have Docker installed and running on the host system. If not, [install Docker](https://docs.docker.com/engine/install/).
+- You have a Docker Hub account. If not, [register for a free Docker Hub account](https://hub.docker.com/signup).
+
+## Step 1: Create a simple Go application
+
+:::note
+If you already have a Go application, you can [proceed to Step 2](#step-2-install-the-dagger-sdk-for-go).
+:::
+
+Create a simple Go application as follows:
+
+1. Create and change into a new directory for the application.
+
+  ```bash
+  mkdir myapp
+  cd myapp
+  ```
+
+1. Declare the module path for your application.
+
+  ```bash
+  go mod init myapp
+  ```
+
+1. Create a new file named `main.go` and add the following code to it. Save the file once done.
+
+  ```go
+  package main
+
+  import "fmt"
+
+  func main() {
+      say := greeting()
+      fmt.Println(say)
+  }
+
+  func greeting() string {
+      return "Hello"
+  }
+  ```
+
+1. Create a new test file named `main_test.go` and add the following code to it. Save the file once done.
+
+  ```go
+  package main
+
+  import (
+    "testing"
+
+    "gotest.tools/v3/assert"
+  )
+
+  func TestGreeting(t *testing.T) {
+    g := greeting()
+    should := "Hello"
+    assert.Equal(t, should, g)
+  }
+  ```
+
+1. Run the application by executing the command below.
+
+  ```bash
+  go run main.go
+  ```
+
+  The application displays the string `Hello`.
+
+## Step 2: Install the Dagger SDK for Go
+
+{@include: ../../partials/_install-sdk-go.md}
+
+## Step 3: Run the application with Dagger
+
+You can now begin using the Dagger SDK to build, run, test and deploy your application. This step explains the simplest of these operations: run your application using the Dagger SDK.
```

Comment:
At this point I am pretty confused. My earlier corrections about "tool" vs "application" assumed this guide is helping write my first tool. With perhaps a sample app involved (something to build, test or deploy).

But it appears that both tool and application are mixed together in this guide?

If that's the case, I think we should avoid this. It's confusing to the end user because it doesn't set expectation for the real world, where tool and application will *definitely* be separate.

Sorry if I missed something here.